docs(ses): add doc for legacyRegeneratorRuntimeTaming (#2425)
endojs/endo: packages/ses/docs/lockdown.md

To run multi-tenant applications safely, these dependencies must be carefully
fixed or avoided.

## `legacyRegeneratorRuntimeTaming` Options

`regenerator-runtime` is a widely used package in the ecosystem.
It is used to support generators and async functions transpiled to ES5.

The option `legacyRegeneratorRuntimeTaming` is to fix `regenerator-runtime`
from 0.10.5 to 0.13.7.

The `legacyRegeneratorRuntimeTaming` option, when set to `'safe'`, it does nothing.

> When set to `'unsafe-ignore'`, it convert `Iterator.prototype[Symbol.iterator]` to

Reviewer: ```suggestion
When set to `'unsafe-ignore'`, it converts `Iterator.prototype[Symbol.iterator]` to
```

a getter/setter that ignores all assignments to it.

```js
lockdown(); // legacyRegeneratorRuntimeTaming defaults to 'safe'
// or
lockdown({ legacyRegeneratorRuntimeTaming: 'safe' }); // do nothing
// vs
lockdown({ legacyRegeneratorRuntimeTaming: 'unsafe-ignore' }); // try fix compatibility with regenerator-runtime

> Iterator.prototype[Symbol.iterator] = function() { return this } // this assignment fails but not throwing with unsafe-ignore

Reviewer: ```suggestion
Iterator.prototype[Symbol.iterator] = function() { return this } // this assignment fails without throwing with unsafe-ignore
```
